src/Training.AirBnb.Clone.Backend/AirBnb.Api/Controllers/NotificationsController.cs

```diff
@@ -12,63 +14,87 @@ public class NotificationsController : ControllerBase
+        public NotificationsController(
+            IEmailTemplateService entityBaseService, 
+            IEmailService emailService, 
+            IEmailManagementService emailManagementService,
+            IMapper mapper)
         {
             _emailTemplateService = entityBaseService;
             _emailService = emailService;
             _emailManagementService = emailManagementService;
+            _mapper = mapper;
         }
-
+        
+        #region SendEmail
         [HttpPost("emailManagement{userId:guid}/{templateId:guid}")]
         public async Task<IActionResult> SendEmail(Guid userId, Guid templateId)
             => Ok(await _emailManagementService.SendEmailAsync(userId, templateId));
+        #endregion
         
-        
+        #region GetAllTemplates
         [HttpGet("emailTemplates")]
         public IActionResult GetAllTemplates()
         {
-            var result = _emailTemplateService.Get(emailtemplate => true).ToList();
+            var result = _emailTemplateService.Get(emailTemplate => true)
+                .Select(et => _mapper.Map<EmailTemplateDto>(et)).ToList();
             return result.Any() ? Ok(result) : NoContent();
         }
-
+        #endregion
+        
+        #region GetByTemplateId
         [HttpGet("emailTemlates/emailTemplateId:guid")]
-        public async Task<IActionResult> GetByTemplateId(Guid id)
-            => Ok(await _emailTemplateService.GetByIdAsync(id));
+        public async Task<IActionResult> GetByTemplateId(Guid templateId)
+            => Ok(_mapper.Map<EmailTemplateDto>(await _emailTemplateService.GetByIdAsync(templateId)));
+        #endregion
         
+        #region AddTemplate
         [HttpPost("emailTemplates")]
-        public async Task<IActionResult> AddTemplate([FromBody] EmailTemplate emailTemplate)
-           =>  Ok(await _emailTemplateService.CreateAsync(emailTemplate));
-
+        public async Task<IActionResult> AddTemplate([FromBody] EmailTemplateDto emailTemplate)
+           =>  Ok(_mapper.Map<EmailTemplateDto>(await _emailTemplateService.CreateAsync(_mapper.Map<EmailTemplate>(emailTemplate))));
+        #endregion
+        
+        #region UpdataTemplate
         [HttpPut("emailTemplates")]
-        public async Task<IActionResult> UpdateTemplate([FromBody] EmailTemplate emailTemplate)
+        public async Task<IActionResult> UpdateTemplate([FromBody] EmailTemplateDto emailTemplate)
         {
-            Ok(await _emailTemplateService.UpdateAsync(emailTemplate));
+            Ok(_mapper.Map<EmailTemplateDto>(await _emailTemplateService.UpdateAsync(_mapper.Map<EmailTemplate>(emailTemplate))));
             return NoContent();
         }
-
+        #endregion
+        
+        #region DeleteTemplate
         [HttpDelete("emailTemplate/{emailTemplateId:guid}")]
         public async Task<IActionResult> DeleteTemplate([FromRoute] Guid emailTemplateId)
         {
             Ok(await _emailTemplateService.DeleteAsync(emailTemplateId));
```

Comment:
controller ichida methodlarni Ok() ga o'rash shart emas. Faqatgina return bo'layotgan narsani Ok() ga o'rash to'g'ri bo'ladi.